### PR TITLE
Delete owner feature

### PR DESCRIPTION
**Pull request by ahmadkhaledCS**
Added a new endpoint that removes an owner from the database.

### CODE REVIEW

**Comment by arey** [comment]
Hi @ahmadkhaledCS what are your motivation to add this endpoint? I mean the UI does not use it. It looks like dead code.

File: spring-petclinic-customers-service/src/main/java/org/springframework/samples/petclinic/customers/web/OwnerResource.java [hunk 1]
```diff
@@ -87,4 +87,18 @@ public void updateOwner(@PathVariable("ownerId") @Min(1) int ownerId, @Valid @Re
...
+     * Remove an Owner
+     */
+    @DeleteMapping("/delete")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteOwner(@RequestParam @Min(1) int ownerId){
+        if(ownerRepository.existsById(ownerId)){
+            ownerRepository.delete(ownerRepository.getReferenceById(ownerId));
+            log.info("Owner removed");
+        }else {
+            log.info("Owner does not exist");
```
Comment (arey) [inline spring-petclinic-customers-service/src/main/java/org/springframework/samples/petclinic/customers/web/OwnerResource.java:101]: Maybe throw a 404?

File: spring-petclinic-customers-service/src/main/java/org/springframework/samples/petclinic/customers/web/OwnerResource.java [hunk 2]
```diff
@@ -87,4 +87,18 @@ public void updateOwner(@PathVariable("ownerId") @Min(1) int ownerId, @Valid @Re
         log.info("Saving owner {}", ownerModel);
         ownerRepository.save(ownerModel);
     }
+
+    /**
+     * Remove an Owner
+     */
+    @DeleteMapping("/delete")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteOwner(@RequestParam @Min(1) int ownerId){
```
Comment (arey) [inline spring-petclinic-customers-service/src/main/java/org/springframework/samples/petclinic/customers/web/OwnerResource.java:96]: A unit test could be fine :D

**Comment by ahmadkhaledCS** [comment]
Hi @arey , This is my first time continuing to open a source. I was just trying it out, and i found your project interesting. If you want, you can list some changes for me to do, or you can decline the change

**Pull request closed without merging**